modules/create_timepoint/jsx/createTimepointIndex.js

```diff
@@ -0,0 +1,399 @@
+            state.form.options.languages = data.languages;
+            state.form.display.languages = true;
+          }
+          this.setState(state);
+        });
+      } else {
+        response.json().then((data) => {
+          if (data.error) {
+            // display conflicts on form.
+            this.setState({messages: [data.error]});
+          }
+        });
+        console.error(response.statusText);
+      }
+    }).catch((error) => {
+      const state = Object.assign({}, this.state);
+      state.messages = [{message: error}];
+      this.setState(state);
+    });
+  }
+
+  /**
+   * Visit Labels refreshes when Subproject changes.
+   */
+  handleVisitLabel() {
+    const state = Object.assign({}, this.state);
+    state.form.options.visit = state.storage.visit[
+      state.form.value.subproject
+      ];
+    if (state.form.value.subproject) {
+      state.form.value.visit = Object.keys(state.storage.visit[
+        state.form.value.subproject
+        ])[0];
+    }
+    state.form.display.visit = true;
+    this.setState(state);
+  }
+
+  /**
+   * Set the form data based on state values of child elements/components
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setForm(formElement, value) {
+    const state = Object.assign({}, this.state);
+    state.form.value[formElement] = value;
+    this.setState(state);
+    if (formElement === 'subproject') {
+      this.handleVisitLabel();
+    }
+  }
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+    const state = Object.assign({}, this.state);
+    const url = `${this.props.baseURL}/create_timepoint/Timepoint`;
+    const send = JSON.stringify({
+      candID: state.url.params.candID,
+      identifier: state.url.params.identifier,
+      subproject: state.form.value.subproject,
+      psc: state.form.value.psc,
+      project: state.form.value.project,
+      visit: state.form.value.visit,
+      languages: state.form.value.languages,
+    });
+    fetch(
+      url, {
+        method: 'POST',
+        credentials: 'same-origin',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: send,
+      }
+    ).then((response) => {
+      if (response.ok) {
+        swal.fire('Success!', 'Time Point created.', 'success')
+          .then(() => {
+            window.location.replace(
+              `${this.props.baseURL}/${this.state.url.params.candID}`
+            );
+          });
+      } else {
+        response.json().then((data) => {
+          if (data.error) {
+            // display conflicts on form.
+            this.setState({messages: JSON.parse(data.error)});
+          }
+        });
+      }
+    }).catch((error) => {
+      console.error('Error! ' + error);
+    });
+  }
+
+  /**
+   * @return {DOMRect}
```

Comment:
I don't know what a "DOMRect" is and this is missing a comment.

Reply:
thanks @driusan, see latest commits